maap/Secrets.py

```diff
@@ -0,0 +1,129 @@
+import requests
+import logging
+import json
+from maap.utils import endpoints
+from maap.utils import requests_utils
+
+
+class Secrets:
+    """
+    Functions used for Member secrets API interfacing
+    """
+    def __init__(self, config, api_header):
+        self._api_header = api_header
+        # self._members_endpoint = members_endpoint
+        self._members_endpoint = "https://api.dit.maap-project.org/api/members/self"
+        self._logger = logging.getLogger(__name__)
+
+
+    def get_secrets(self):
+        """
+        Returns a list of secrets for a given user.
+
+        Returns:
+            list: Secret names for a given user.
+        """
+        try:
+            response = requests.get(
+                url = f"{self._members_endpoint}/{endpoints.MEMBERS_SECRETS}",
+                headers=self._api_header
+            )
+
+            return json.loads(response.text)
+
+        except Exception as ex:
+            raise(f"Error retrieving secrets: {ex}")
+
+
+    def get_secret(self, secret_name=None):
+        """
+        Returns secret value for provided secret name.
+
+        Args:
+            secret_name (str, required): Secret name.
+            secret_value (str, optional): Secret value.
+
+        Returns:
+            dict: Secret name and value.
+
+        Raises:
+            ValueError: If secret name is not provided.
+        """
+        try:
+            if secret_name is None:
+                raise ValueError("Failed to get secret value. Please provide secret name.")
+
+            response = requests.get(
+                url = f"{self._members_endpoint}/{endpoints.MEMBERS_SECRETS}/{secret_name}",
+                headers=self._api_header
+            )
+
+            return json.loads(response.text)
```

Comment:
I think this should just return a value and not a dict.
Returning a dict object can complicate users code using maag.py
For eg.
```
token = maap.Secrets().get_secret("name").get("name")
```

vs if we return just the value it could look like 
```
token = maap.Secrets().get_secret("name")
```

Reply:
Updated to support this.